bin/app.ts

```diff
@@ -96,47 +110,137 @@ class ServiceDeployIAM extends cdk.Stack {
-                         "logs:CreateLogGroup",
-                         "logs:DescribeLogGroup",
-                         "logs:DeleteLogGroup",
-                         "logs:CreateLogStream",
-                         "logs:DescribeLogStreams",
-                         "logs:DeleteLogStream",
-                         "logs:FilterLogEvents"
+                         "autoscaling:Describe*",
+                         "cloudwatch:*",
+                         "logs:*",
+                         "sns:*",
+                         "iam:GetPolicy",
+                         "iam:GetPolicyVersion",
+                         "iam:GetRole"
+                    ]
+               })
+          );
+          // Lambda Role Creation
+          serviceRole.addToPolicy(
+               new PolicyStatement({
+                    effect: Effect.ALLOW,
+                    resources: ["*"],
+                    actions: [            
+                        "events:*"
                     ]
                })
           );
 
+          serviceRole.addToPolicy(
+               new PolicyStatement({
+                    effect: Effect.ALLOW,
+                    resources: [`arn:aws:iam::${accountId}:role/horizon*`],
+                    actions: [            
+                        "iam:*"
+                    ]
+               })
+          );
+
+
+          // SQS Full Access
+          serviceRole.addToPolicy(
+               new PolicyStatement({
+                    effect: Effect.ALLOW,
+                    resources: ["*"],
+                    actions: [            
+                         "sqs:*"
+                    ]
+               })
+          );
 
           // Lambda policy
           serviceRole.addToPolicy(
                new PolicyStatement({
                     effect: Effect.ALLOW,
                     resources: lambdaResources,
                     actions: [            
-                         "lambda:GetFunction",
-                         "lambda:CreateFunction",
-                         "lambda:DeleteFunction",
-                         "lambda:UpdateFunctionConfiguration",
-                         "lambda:UpdateFunctionCode",
-                         "lambda:ListVersionsByFunction",
-                         "lambda:PublishVersion",
-                         "lambda:CreateAlias",
-                         "lambda:DeleteAlias",
-                         "lambda:UpdateAlias",
-                         "lambda:GetFunctionConfiguration",
-                         "lambda:AddPermission",
-                         "lambda:RemovePermission",
-                         "lambda:InvokeFunction"
+                         "cloudformation:DescribeStacks",
+                         "cloudformation:ListStackResources",
+                         "cloudwatch:ListMetrics",
+                         "cloudwatch:GetMetricData",
+                         "ec2:DescribeSecurityGroups",
+                         "ec2:DescribeSubnets",
```

Comment:
These permissions won't work.

They are targeting the `lambdaResources`.

Lambda resources do not have an action called `ec2:DescribeSubnets` for example.